modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -154,7 +168,7 @@ object Run extends ScalaCommand[RunOptions] {
       builds.main match {
         case s: Build.Successful =>
           val (process, onExit) = maybeRun(s, allowTerminate = true)
-            .orExit(logger)
+            .orExit(logger).getOrElse(sys.exit(1))
```

Comment:
The `sys.exit` is for when we list main classes, right? A `0` exit code would work fine I think (users requested the main class list themselves, listing them isn't really an error).

Reply:
Oh, it seems it's not supposed to be called anyway (as `allowTerminate` is true). A `sys.error("Can't happen")` would have done the job then (and be more helpful if ever it ends up being called anyway), but it's no big deal.